script.js

```diff
@@ -0,0 +1,244 @@
+            <a href="${workArray.liveLink}">
+              <button>
+                <p>See live</p> <img src="./Icons/social/Icon - Export.svg" alt="">
+              </button>
+            </a>
+            <a href="${workArray.sourceLink}">
+              <button>
+                <p>See source</p> <img src="./Icons/social/Frame.svg" alt="">
+              </button>
+            </a>
+          </div>
+        </div>
+      </div>
+    </div>`;
+
+      modal.classList.remove('hidden');
+      modal.innerHTML = modalHtml;
+
+      const closePopup = document.querySelector('.closePopup');
+      closePopup.addEventListener('click', () => {
+        modal.classList.add('hidden');
+      });
+    });
+  });
+}
+
+const showbtn = document.querySelector('.active');
+const hidebtn = document.querySelector('.navbar');
+const popup = document.querySelector('.project-detail');
+
+const x = document.getElementById('menubar');
+const show = () => {
+  x.classList = 'active';
+};
+
+const hide = () => {
+  x.classList = 'navbar';
+};
+
+const remove = () => {
+  popup.classList.remove('hidden');
+};
+
+showbtn.addEventListener('click', show);
+hidebtn.addEventListener('click', hide);
+popup.addEventListener('click', remove);
+
+// contact form
+
+function showPopup() {
+  // const form = document.getElementById('contact-form');
+  // const emailInput = form.querySelector('input[name="email"]');
+  // const errorMessage = document.getElementById('error-message');
+
+  // form.addEventListener('submit', (event) => {
+  //   event.preventDefault();
+
+  //   const email = emailInput.value;
+
+  //   if (email.toLowerCase() === email) {
+  //     form.submit();
+  //   } else {
+  //     errorMessage.textContent = 'Please enter the email address in lowercase.';
+  //     errorMessage.style.display = 'block';
+  //   }
+  // });
+
+  const form = document.getElementById('contact-form');
+  const emailInput = form.querySelector('input[name="email"]');
+  const errorMessage = document.getElementById('error-message');
+
+  window.addEventListener('DOMContentLoaded', () => {
+    const formData = localStorage.getItem('formData');
+    if (formData) {
+      const { firstname, email, msg } = JSON.parse(formData);
+      form.firstname.value = firstname;
+      emailInput.value = email;
+      form.msg.value = msg;
+    }
+  });
+
+  form.addEventListener('change', () => {
+    const firstname = form.firstname.value;
+    const email = emailInput.value;
+    const msg = form.msg.value;
+
+    const formData = { firstname, email, msg };
+    localStorage.setItem('formData', JSON.stringify(formData));
+  });
+
+  form.addEventListener('submit', (event) => {
+    event.preventDefault();
+
+    const email = emailInput.value;
+
+    if (email.toLowerCase() === email) {
+      form.submit();
+    } else {
+      errorMessage.textContent = 'Please enter the email address in lowercase.';
+      errorMessage.style.display = 'block';
+    }
+  });
+}
```

Comment:
- [ ] Well done using `localStorage.getItem` and `localStorage.setItem` however it is required that the form data for all inputs should be stored in a single object, saved to a single local storage registry.
````
const formData = {
    name: nameInput.value,
    email: emailInput.value,
    body: bodyInput.value,
};
````

- [ ] It is required too that :
- The contact form information is persistent to page reloads.
- After filling the form, and not submitting it, the entered values should be saved in the browser.
Which is not happening when filling the form in your project. Kindly fix the issue.